workers/unity/Assets/Playground/Scripts/MonoBehaviours/CheckForPlayerCollision.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Improbable.Gdk.Core.GameObjectRepresentation;
+using UnityEngine;
+using Collisions = Generated.Playground.Collisions;
+using Empty = Generated.Playground.Empty;
+
+public class CheckForPlayerCollision : MonoBehaviour
+{
+    [Require] private Collisions.Writer writer;
+
+    void OnTriggerEnter(Collider other)
```

Comment:
This actually gets called regardless of whether you are enabled or not so you need to null check the writer. Currently this works as you happen to get authority on the same tick as adding the entity, and this is called after injection.

Reply:
Didn't know it's called when disabled, will fix